docs/backend/README.md

```diff
@@ -1,3 +1,104 @@
+The root of our backend application is in `src/backend`. noteworthy subdirectories are annotated here:
+- `aspen` - Most of our backend application lives here
+  - `api` - entrypoint for our backend API service
+    - `schemas` - API input/output validation models live here
+    - `utils` - Code used by multiple endpoints (this could be better organized)
+    - `views` - API endpoint code lives here
+  - `workflows` - code related to our compute jobs
+  - `database` - code related to establishing DB connections / sessions
+    - `models` - SQLAlchemy models used across our backend application
+- `database_migrations` - alembic migrations
+- `etc` - some basic setup configuration for our backend container
+
+## Discoverability
+Our backend API is self-documenting via OpenAPI and JSON-Schema:
+- API documentation: https://api.czgenepi.org/v2/docs
+- JSON Schema: https://api.czgenepi.org/v2/openapi.json
+
+## Creating a new API endpoint.
+Some endpoints will only need the last few steps, but this is the maximal case:
+1. Create a new module in `src/backend/aspen/api/views` and add a router:
+```python
+# src/backend/aspen/api/views/new_module.py
+from fastapi import APIRouter
+router = APIRouter()
+```
+2. Import it into `src/backend/aspen/api.main.py` and create a route for it:
+```python
+# src/backend/aspen/api/main.py
+import new_module from api.views
+...
+    _app.include_router(
+        new_module.router,
+        prefix="/v2/new_module",
+        dependencies=[Depends(get_auth_user)],  # If all endpoints require authentication
+    )
+```
+3. Add any new pydantic validation schemas as necessary:
+```python
+# src/backend/aspen/api/schemas/new_module.py
+from aspen.api.schemas.base import BaseResponse, BaseRequest
+
+
+class NewRequest(BaseRequest):
+    foo: str
+
+class NewResponseItem(BaseResponse):
+    id: int
+    bar: str
+
+class NewResponseList(BaseResponse):
+    rows: List[NewResponseItem]
+```
+4. Add endpoints to your view module.
+```python
+# src/backend/aspen/api/schemas/new_schemas.py
+import sqlalchemy as sa
+from fastapi import APIRouter, Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from aspen.api.auth import get_auth_user
+from aspen.api.deps import get_db, get_settings
+from aspen.api.schemas.new_module import NewRequest, NewResponseList
+from aspen.api.settings import Settings
+from aspen.database.models import User, SomeModel
+
+router = APIRouter()
+
+
+# Specify the HTTP method, sub-path (appending to the path added to main.py) and response model type.
+# The response model type here populates our API documentation.
+@router.get("/", response_model=NewResponseList) # All endpoints that return a list must end with a trailing slash
+async def list_items(
+    request: NewRequest, # GET requests often don't need input model validation, but this is here for completeness.
+    db: AsyncSession = Depends(get_db),
+    settings: Settings = Depends(get_settings),
+    user: User = Depends(get_auth_user),
```

Comment:
I think this line deserves a special call out. While `user` does not actually get used below, the call to `get_auth_user` is actually the hinge of our auth right now. It being called is what prevents access by unauthenticated users. I don't think that's particularly obvious, but since it's a core part of our auth process it's important to make it explicit.

Unlike the `settings` above, which is just serving as an example of how you could pull the app settings if you needed them in your view function, the `Depends(get_auth_user)` is doing more than just pulling the user's data: it's also kicking a 401 if the request is from a non-authenticated user and blocking the view code from firing.

Reply:
fair enough, I've added a comment block with more detail.

Reply:
Perfect, that's extremely clear. Thanks!